packages/datetime/src/dateInput.tsx

```diff
@@ -362,6 +361,10 @@ export class DateInput extends AbstractPureComponent<IDateInputProps, IDateInput
     };
 
     private handleInputClick = (e: React.SyntheticEvent<HTMLInputElement>) => {
+        // stop propagation to the Popover's internal handleTargetClick handler;
+        // otherwise, the popover will flicker closed as soon as it opens.
+        e.stopPropagation();
```

Comment:
strange, how come this wasn't necessary before?

Reply:
Not sure, but `DateRangeInput` already had something like this FWIW:
https://github.com/palantir/blueprint/blob/develop/packages/datetime/src/dateRangeInput.tsx#L620